src/jit/valuenum.cpp

```diff
@@ -256,87 +256,123 @@ TFp FpRem(TFp dividend, TFp divisor)
     return (TFp)fmod((double)dividend, (double)divisor);
 }
 
-VNFunc GetVNFuncForOper(genTreeOps oper, bool isUnsigned)
+//--------------------------------------------------------------------------------
+// GetVNFuncForOper:     - Given a genTreeOper this function Returns the correct
+//                         VNFunc to use for ValueNumbering
+//
+// Arguments:
+//    oper               - The gtOper value from the GenTree node
+//    unsignedOper       - 'true' if ww need to perform the unsigned variant of 'oper'
+//    overflowChecking   - 'true' if this 'oper' will also perform am overflow check
+//
+// Return Value:
+//               - The VNFunc to used for this kind of GenTree nodea
+//
+// Notes:        - Not all combinations are legal.  An assert will fire
+//                 when unsignedOper or overflowCheckingOper is set for
+//                 an oper that does not support that combination.
+//
+VNFunc GetVNFuncForOper(genTreeOps oper, bool unsignedOper, bool overflowChecking)
 {
-    if (!isUnsigned || (oper == GT_EQ) || (oper == GT_NE))
-    {
-        return VNFunc(oper);
-    }
-    switch (oper)
+    VNFunc result = VNF_COUNT; // An illegal value
+
+    // For most genTreeOpers we just use the VNFunc with the same enum value as the oper
+    //
+    if ((!unsignedOper && !overflowChecking) || (oper == GT_EQ) || (oper == GT_NE))
```

Comment:
It seems a bit strange that for GT_EQ/NE `overflowChecking` is ignored while for all other relops it is asserted that it is false.

Reply:
Yes,  noticed that, I can work in a fix for this (and unsignedOper as well)